src/fragments/lib/push-notifications/android/record_notifications/20_record_notification_received.mdx

```diff
@@ -0,0 +1,114 @@
+        // First remove any parameters that shouldn't be passed to the app
+        // * The wakelock ID set by the WakefulBroadcastReceiver
+        data.remove("androidx.content.wakelockid")
+        // Convert the RemoteMessage into a PinpointNotificationPayload
+        PinpointNotificationPayload notificationPayload = PinpointNotificationPayload.createFromRemoteMessage(new RemoteMessage(data));
+
+        // Record notification received with Amplify
+        Amplify.Notifications.Push.recordNotificationReceived(notificationPayload,
+            result -> Log.i("MyAmplifyApp", "Successfully recorded a notification received"),
+            error -> Log.e("MyAmplifyApp", "Error recording notification received", error)
+        );
+    }
+}
+```
+
+</Block>
+<Block name="Kotlin - Callbacks">
+
+```kotlin
+class MyAppService : FirebaseMessagingService() {
+    override fun handleIntent(intent: Intent?) {
+
+        // TODO(developer): Handle FCM messages here
+
+        val data = intent?.extras ?: Bundle()
+        // First remove any parameters that shouldn't be passed to the app
+        // * The wakelock ID set by the WakefulBroadcastReceiver
+        data.remove("androidx.content.wakelockid")
+        // Convert the RemoteMessage into a PinpointNotificationPayload
+        val notificationPayload = PinpointNotificationPayload.createFromRemoteMessage(RemoteMessage(data))
+
+        // Record notification received with Amplify
+        Amplify.Notifications.Push.recordNotificationReceived(notificationPayload,
+            { Log.i("MyAmplifyApp", "Successfully recorded a notification received") },
+            { Log.e("MyAmplifyApp", "Error recording notification received", error) }
+        )
+    }
+}
+```
+
+</Block>
+<Block name="Kotlin - Coroutines">
+
+```kotlin
+class MyAppService : FirebaseMessagingService() {
+   override fun handleIntent(intent: Intent?) {
+
+        // TODO(developer): Handle FCM messages here
+
+        val data = intent?.extras ?: Bundle()
+        // First remove any parameters that shouldn't be passed to the app
+        // * The wakelock ID set by the WakefulBroadcastReceiver
+        data.remove("androidx.content.wakelockid")
+        // Convert the RemoteMessage into a PinpointNotificationPayload
+        val notificationPayload = PinpointNotificationPayload.createFromRemoteMessage(RemoteMessage(data))
+
+        try {
+            // let Amplify handle foreground and background message received
+            Amplify.Notifications.Push.recordNotificationReceived(remoteMessage)
+            Log.i("MyAmplifyApp", "Successfully recorded a notification received")
+        } catch (error: PushNotificationsException) {
+            Log.e("MyAmplifyApp", "Error recording notification received", error)
+        }
+    }
+}
+```
+
+</Block>
+<Block name="RxJava">
+
+```java
+class MyAppService extends FirebaseMessagingService() {
+    public void handleIntent(Intent intent) {
+
```

Comment:
Todo: Add instructions to setup activity deep linking. https://docs.amplify.aws/sdk/push-notifications/setup-push-service/q/platform/android/#open-a-deep-link

Reply:
Also add a callout that recordNotificationOpened is called internally when opening deep link activity.

Reply:
I wonder if this is something we need to provide instructions for when deep linking is part of general Android documentation? https://developer.android.com/training/app-links/deep-linking